backend/controllers/history.controller.js

```diff
@@ -0,0 +1,82 @@
+import historyVideos from '../models/history.model.js'
+
+//@route   GET api/history
+//@desc    Get All history Videos
+//@access  Private
+const getAllHistoryVideos = async (req, res) => {
+    try {
+        const id = req.user;
+        const videos = await historyVideos.findOne({ user: id })
```

Comment:
to avoid null checks on the frontend, you can send an empty array from here if the videos variable is null.

Reply:
If we don't have any videos in model then model field named historyvideos will automatically be empty array as its type is Array in model. So I don't think here null check is required.
Hope that's okay :)